### PR TITLE
Feature/filtering

### LINKED ISSUE #41: As a gamer, I would like to filter games so that I can find more fitting games
- [x] Filter menu

Options to filter on:
- [x] Filter platform
- [x] Filter playtime
- [x] Filter release date
- [x] Filter amount of players

### PR DESCRIPTION
**Pull request by x183**
Added filtering, fixes #41

### CODE REVIEW

**Review by GAsplund — APPROVED** [review]
LGTM!

File: backend/src/routers/gameRouter.ts [hunk 1]
```diff
@@ -151,6 +153,75 @@ gameRouter.post(
...
+			gte: new Date(body.releaseAfter)
+		};
+	if (body.playerCount) {
+		filter.playerMax = { gte: body.playerCount };
+		filter.playerMin = { lte: body.playerCount };
+	}
+	if (body.platform)
+		filter.platform = { name: body.platform };
+	if (body.playtime)
+		filter.playtime = body.playtime;
```
[the remark this answers is not in the document]
Comment (x183) [reply backend/src/routers/gameRouter.ts]: yes

**Comment by x183** [comment]
I fixed the date and number issues for filtering; however, we might want to create a more general fix since as of now the number and date fields when adding games/game suggestions suffer from the same issue :shrug:

File: frontend/src/components/GamesList/GamesList.tsx [hunk 2]
```diff
@@ -17,55 +19,91 @@ interface Game {
...
+			if (e.target.value) searchFilter.name = e.target.value;
 		}
-	}, 300)
+		if (platform) searchFilter.platform = platform;
+		if (releaseBefore) searchFilter.releaseBefore = new Date(releaseBefore);
+		if (releaseAfter) searchFilter.releaseAfter = new Date(releaseAfter);
+		if (playtime) searchFilter.playtime = playtime;
+		if (playerCount) searchFilter.playerCount = playerCount;
+		postData(searchFilter);
+		console.log(releaseBefore);
```
Comment (Oscariremma) [inline frontend/src/components/GamesList/GamesList.tsx:53]: Remove console log
```suggestion
```

Comment (x183) [reply frontend/src/components/GamesList/GamesList.tsx]: yes

**Review by Oscariremma — APPROVED** [review]
Now everything looks good!

**Pull request merged**